packages-ui/roosterjs-react/lib/emoji/components/EmojiPane.tsx

```diff
@@ -0,0 +1,575 @@
+const VerticalDirectionKeys: number[] = [KeyCodes.up, KeyCodes.down];
+const DirectionKeys: number[] = [
+    KeyCodes.left,
+    KeyCodes.right,
+    KeyCodes.up,
+    KeyCodes.down,
+    KeyCodes.home,
+    KeyCodes.end,
+];
+
+const TooltipCalloutProps: ICalloutProps = {
+    isBeakVisible: true,
+    beakWidth: 16,
+    gapSpace: 0,
+    setInitialFocus: true,
+    doNotLayer: false,
+    directionalHint: DirectionalHint.bottomCenter,
+};
+
+/**
+ * @internal
+ * Types of emoji pane size
+ */
+export const enum EmojiPaneMode {
+    Quick,
+    Partial,
+    Full,
+}
+
+/**
+ * @internal
+ * Types of emoji Navigation direction
+ */
+export const enum EmojiPaneNavigateDirection {
+    Horizontal,
+    Vertical,
+}
+
+/**
+ * @internal
+ * Emoji Pane data
+ */
+export interface EmojiPaneState {
+    index: number;
+    mode: EmojiPaneMode;
+    currentEmojiList: Emoji[];
+    currentFamily: EmojiFamilyKeys;
+    search: string;
+    searchInBox: string;
+}
+
+/**
+ * @internal
+ * Emoji Pane customizable data
+ */
+export interface EmojiPaneProps {
+    searchDisabled?: boolean;
+    hideStatusBar?: boolean;
+    navBarProps?: Partial<EmojiNavBarProps>;
+    statusBarProps?: Partial<EmojiStatusBarProps>;
+    emojiIconProps?: Partial<EmojiIconProps>;
+    searchBoxString?: LocalizedStrings<EmojiStringKeys>;
+    classNames?: IProcessedStyleSet<IStyleSet<EmojiPaneStyle>>;
+    onSelect: (emoji: Emoji, wordBeforeCursor: string) => void;
+    baseId: number;
+    strings: Record<string, string>;
+}
+
+/**
+ * @internal
+ * EmojiPane Style classes
+ */
+interface EmojiPaneStyle {
+    quickPicker: IStyleSet;
+    tooltip: IStyleSet;
+    emojiTextInput: IStyleSet;
+    partialList: IStyleSet;
+    fullListContent: IStyleSet;
+    fullListBody: IStyleSet;
+    fullList: IStyleSet;
+    roosterEmojiPane: IStyleSet;
+}
+
+const AriaAttributes = {
+    ActiveDescendant: 'aria-activedescendant',
+    AutoComplete: 'aria-autocomplete',
+    Controls: 'aria-controls',
+    Expanded: 'aria-expanded',
+    HasPopup: 'aria-haspopup',
+    Owns: 'aria-owns',
+    Pressed: 'aria-pressed',
+};
+
+/**
+ * @internal
+ * Emoji pane component functions
+ */
+export interface EmojiPane {
+    navigate: (change: number, direction: EmojiPaneNavigateDirection) => number;
+    getEmojiElementIdByIndex: (index: number) => string;
+    getSelectedEmoji: () => Emoji;
+    showFullPicker: (fullSearchText: string) => void;
+    setSearch: (value: string) => void;
+    normalizeSearchText: (text: string, colonIncluded: boolean) => string;
+    getSearchResult: (searchValue: string, mode: EmojiPaneMode) => Emoji[];
+    getEmojiIconId: (emoji: Emoji) => string;
+}
+
+const EmojiPane = React.forwardRef(function EmojiPaneFunc(
+    props: EmojiPaneProps,
+    ref: React.Ref<EmojiPane>
+) {
+    let searchBox: ITextField;
+
+    let emojiBody: HTMLElement;
+    let input: HTMLInputElement;
+
+    const [state, setState] = React.useState<EmojiPaneState>({
```

Comment:
Need to split the states into each items. e.g.

const [index, setIndex] = React.useState(0);
const [mode, setMode] = React.useState(EmojiPaneMode.Quick);
...